crates/pacdef_core/src/backend/actual/rustup.rs

```diff
@@ -0,0 +1,231 @@
+use crate::backend::backend_trait::{Backend, Switches, Text};
+use crate::backend::macros::impl_backend_constants;
+use crate::{Group, Package};
+use anyhow::{Context, Result};
+use core::panic;
+use std::collections::HashSet;
+use std::os::unix::process::ExitStatusExt;
+use std::process::{Command, ExitStatus};
+
+#[derive(Debug, Clone)]
+pub struct Rustup {
+    pub(crate) packages: HashSet<Package>,
+}
+
+enum Repotype {
+    Toolchain,
+    Component,
+}
+
+const BINARY: Text = "rustup";
+const SECTION: Text = "rustup";
+
+const SWITCHES_INSTALL: Switches = &["component", "add"];
+const SWITCHES_INFO: Switches = &["component", "list", "--installed"];
+const SWITCHES_MAKE_DEPENDENCY: Switches = &[];
+const SWITCHES_NOCONFIRM: Switches = &[];
+const SWITCHES_REMOVE: Switches = &["component", "remove"];
+
+const SUPPORTS_AS_DEPENDENCY: bool = false;
+
+impl Backend for Rustup {
+    impl_backend_constants!();
+
+    fn get_all_installed_packages(&self) -> Result<HashSet<Package>> {
+        let mut toolchains_vec = self
+            .run_toolchain_command(get_info_switches(Repotype::Toolchain))
+            .context("Getting installed toolchains")?;
+
+        let mut toolchains: HashSet<Package> = toolchains_vec
+            .iter()
+            .map(|name| ["toolchain", name].join("/").into())
+            .collect();
+
+        let packages: HashSet<Package> = self
+            .run_component_command(get_info_switches(Repotype::Component), &mut toolchains_vec)
+            .context("Getting installed components")?
+            .iter()
+            .map(|name| ["component", name].join("/").into())
+            .collect();
+
+        toolchains.extend(packages);
+        Ok(toolchains)
+    }
+
+    fn get_explicitly_installed_packages(&self) -> Result<HashSet<Package>> {
+        self.get_all_installed_packages()
+            .context("Getting all installed packages")
+    }
+
+    fn make_dependency(&self, _: &[Package]) -> Result<ExitStatus> {
+        anyhow::bail!("Not supported by {}", self.get_binary())
```

Comment:
Go with `use anyhow::bail`, and replace all usages with `bail!(...)`

Reply:
Done